enterprise/query-logging/src/test/java/org/neo4j/kernel/impl/query/QueryLoggerIT.java

```diff
@@ -312,14 +313,15 @@ public void queryLogRotation() throws Exception
 
         database.shutdown();
 
-        List<String> lines = readAllLines( logFilename );
-        assertTrue( "Expected log file to have at least one log entry", lines.size() >= 1 );
+        File[] queryLogs = fileSystem.get().listFiles( logsDirectory, ( dir, name ) -> name.startsWith( "query.log" ) );
 
-        lines = readAllLines( shiftedLogFilename1 );
-        assertTrue( "Expected shifted log file to have at least one log entry", lines.size() >= 1 );
+        assertThat( "Expect to have more then one query log file.", queryLogs.length, greaterThanOrEqualTo( 2 ) );
 
-        lines = readAllLines( shiftedLogFilename2 );
-        assertTrue( "Expected second shifted log file to have at least one log entry", lines.size() >= 1 );
+        List<String> loggedQueries = Arrays.stream( queryLogs )
+                                        .map( this::readAllLinesSilent )
+                                        .flatMap( Collection::stream )
+                                        .collect( Collectors.toList() );
+        assertTrue( "Expected log file to have at least one log entry", loggedQueries.size() >= 1 );
```

Comment:
Hmm. Do we miss any log statements due to the rotation here? Wouldn't they all end up in the buffer and then sent to the new log file? In that case could assert that `loggedQueries.size() == 100` (the number of queries sent).